### ISSUE TITLE
ORS does not work with OSM from Geofabrik

### ISSUE THREAD

**Issue by neatnerd** [GIScience/openrouteservice#839]
#### Here's what I did
1. I have downloaded osm file for Austria from [here](http://download.geofabrik.de/europe-latest.osm.pbf)
2. Verified that md5 matched
3. Modified docker-compose just to include the file like this:
```
version: '2.4'
services:
  ors-app:
    container_name: ors-app
    ports:
      - 8080:8080
      - 9001:9001
    image: openrouteservice/openrouteservice:latest
    volumes:
      - ./graphs:/ors-core/data/graphs
      - ./elevation_cache:/ors-core/data/elevation_cache
      - ./logs/ors:/var/log/ors
      - ./logs/tomcat:/usr/local/tomcat/logs
      - ./conf:/ors-conf
      - ./austria-latest.osm.pbf:/ors-core/data/osm_file.pbf
    environment:
      - BUILD_GRAPHS=True  # Forces the container to rebuild the graphs, e.g. when PBF is changed
      - "JAVA_OPTS=-Djava.awt.headless=true -server -XX:TargetSurvivorRatio=75 -XX:SurvivorRatio=64 -XX:MaxTenuringThreshold=3 -XX:+UseG1GC -XX:+ScavengeBeforeFullGC -XX:ParallelGCThreads=4 -Xms1g -Xmx2g"
      - "CATALINA_OPTS=-Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=9001 -Dcom.sun.management.jmxremote.rmi.port=9001 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Djava.rmi.server.hostname=localhost"
```
4. Ran docker compose up

---
#### Here's what I got
1. The error seems to be related to the OSM file:
```
2020-12-10 18:37:05,940 ERROR [routing.RoutingProfileManager] - Failed to initialize RoutingProfileManager instance.
java.util.concurrent.ExecutionException: java.lang.RuntimeException: Couldn't process file data/osm_file.pbf, error: java.net.SocketTimeoutException: Read timed out
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?]
	at java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:?]
	at org.heigit.ors.routing.RoutingProfileManager.initialize(RoutingProfileManager.java:164) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfileManager.getInstance(RoutingProfileManager.java:59) ~[classes/:6.3.1]
	at org.heigit.ors.servlet.listeners.ORSInitContextListener.lambda$contextInitialized$0(ORSInitContextListener.java:40) ~[classes/:6.3.1]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.RuntimeException: Couldn't process file data/osm_file.pbf, error: java.net.SocketTimeoutException: Read timed out
	at com.graphhopper.reader.osm.OSMReader.writeOsm2Graph(OSMReader.java:326) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.readGraph(OSMReader.java:178) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.importData(GraphHopper.java:735) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.readData(GraphHopper.java:714) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.process(GraphHopper.java:701) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:680) ~[graphhopper-core-v0.13.9.jar:?]
	at org.heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.importOrLoad(ORSGraphHopper.java:159) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfile.initGraphHopper(RoutingProfile.java:191) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfile.<init>(RoutingProfile.java:135) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:35) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:21) ~[classes/:6.3.1]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	... 1 more
Caused by: java.lang.RuntimeException: java.net.SocketTimeoutException: Read timed out
	at com.graphhopper.reader.dem.AbstractTiffElevationProvider.downloadFile(AbstractTiffElevationProvider.java:159) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.reader.dem.AbstractTiffElevationProvider.getEle(AbstractTiffElevationProvider.java:121) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.reader.dem.MultiSourceElevationProvider.getEle(MultiSourceElevationProvider.java:52) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.getElevation(OSMReader.java:715) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at org.heigit.ors.routing.graphhopper.extensions.ORSOSMReader.getElevation(ORSOSMReader.java:433) ~[classes/:6.3.1]
	at com.graphhopper.reader.osm.OSMReader.addNode(OSMReader.java:676) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.processNode(OSMReader.java:641) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.writeOsm2Graph(OSMReader.java:293) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.readGraph(OSMReader.java:178) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.importData(GraphHopper.java:735) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.readData(GraphHopper.java:714) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.process(GraphHopper.java:701) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:680) ~[graphhopper-core-v0.13.9.jar:?]
	at org.heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.importOrLoad(ORSGraphHopper.java:159) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfile.initGraphHopper(RoutingProfile.java:191) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfile.<init>(RoutingProfile.java:135) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:35) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:21) ~[classes/:6.3.1]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	... 1 more
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:252) ~[?:?]
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:292) ~[?:?]
	at java.io.BufferedInputStream.read(BufferedInputStream.java:351) ~[?:?]
	at sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:754) ~[?:?]
	at sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:689) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1615) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520) ~[?:?]
	at com.graphhopper.util.Downloader.fetch(Downloader.java:74) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.util.Downloader.downloadFile(Downloader.java:114) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.reader.dem.AbstractTiffElevationProvider.downloadFile(AbstractTiffElevationProvider.java:155) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.reader.dem.AbstractTiffElevationProvider.getEle(AbstractTiffElevationProvider.java:121) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.reader.dem.MultiSourceElevationProvider.getEle(MultiSourceElevationProvider.java:52) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.getElevation(OSMReader.java:715) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at org.heigit.ors.routing.graphhopper.extensions.ORSOSMReader.getElevation(ORSOSMReader.java:433) ~[classes/:6.3.1]
	at com.graphhopper.reader.osm.OSMReader.addNode(OSMReader.java:676) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.processNode(OSMReader.java:641) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.writeOsm2Graph(OSMReader.java:293) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.reader.osm.OSMReader.readGraph(OSMReader.java:178) ~[graphhopper-reader-osm-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.importData(GraphHopper.java:735) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.readData(GraphHopper.java:714) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.process(GraphHopper.java:701) ~[graphhopper-core-v0.13.9.jar:?]
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:680) ~[graphhopper-core-v0.13.9.jar:?]
	at org.heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.importOrLoad(ORSGraphHopper.java:159) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfile.initGraphHopper(RoutingProfile.java:191) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfile.<init>(RoutingProfile.java:135) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:35) ~[classes/:6.3.1]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:21) ~[classes/:6.3.1]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	... 1 more
```
2. Query of the endpoint returns not ready:
```
{"status":"not ready"}
```
---
#### Here's what I was expecting

I followed the instruction and expected the application to work for the selected osm file

---
#### Here's what I think could be improved
I believe that issue is related to the version of the graphhopper used. The project uses `v0.13.9`, while the latest version is ca. 2.

**Comment by phil294** [comment]
I can confirm this issue. Neither can http://download.openstreetmap.fr/extracts/europe/austria.osm.pbf be read. Not sure where to go from here, so I cant really use ORS at all :-/

**Comment by nilsnolde** [comment]
Not a ORS issue, it's a read timeout on a network request when downloading elevation, see line
`	at org.heigit.ors.routing.graphhopper.extensions.ORSOSMReader.getElevation(ORSOSMReader.java:433) ~[classes/:6.3.1]`

here is the fix: https://ask.openrouteservice.org/t/timeout-exception-on-elevation-data/2345/2

**Comment by neatnerd** [comment]
I have mounted conf folder and edit `app.config` to disabled elevation everywhere (every profile). It still fails with the same error

**Comment by nilsnolde** [comment]
Ok, then two things:

- there is also a global app.config parameter for elevation (not profile-specific), did you set that one too? 
- did you exec into the container and verified that your app.config is taken into account? It’s somewhere in the tomcat/webapp/ors folder 

Sorry, on mobile right now.. If you still can’t build then change the elevation provider in the config to gmted, that should work.

**Comment by neatnerd** [comment]
I am not sure why you point to the tomcat directory, in documentation it says that config is at `/ors-conf`. I exec'ed into running container and yes the config was there. Regarding global variable, I am not sure what you refer to, but this is the entire config file I used:

```
{
  "ors": {
    "info": {
      "base_url": "https://openrouteservice.org/",
      "support_mail": "support@openrouteservice.org",
      "author_tag": "openrouteservice",
      "content_licence": "LGPL 3.0"
    },
    "services": {
      "matrix": {
        "enabled": true,
        "maximum_routes": 100,
        "maximum_routes_flexible": 25,
        "maximum_search_radius": 5000,
        "maximum_visited_nodes": 100000,
        "allow_resolve_locations": true,
        "attribution": "openrouteservice.org, OpenStreetMap contributors"
      },
      "isochrones": {
        "enabled": true,
        "maximum_range_distance": [
          {
            "profiles": "any",
            "value": 50000
          },
          {
            "profiles": "driving-car, driving-hgv",
            "value": 100000
          }
        ],
        "maximum_range_time": [
          {
            "profiles": "any",
            "value": 18000
          },
          {
            "profiles": "driving-car, driving-hgv",
            "value": 3600
          }
        ],
        "fastisochrones": {
          "maximum_range_distance": [
            {
              "profiles": "any",
              "value": 50000
            },
            {
              "profiles": "driving-car, driving-hgv",
              "value": 500000
            }
          ],
          "maximum_range_time": [
            {
              "profiles": "any",
              "value": 18000
            },
            {
              "profiles": "driving-car, driving-hgv",
              "value": 10800
            }
          ],
          "profiles": {
            "default_params": {
              "enabled": false,
              "threads": 12,
              "weightings": "recommended",
              "maxcellnodes": 5000
            },
            "profile-vehicles-hgv": {
              "enabled": true,
              "threads": 12,
              "weightings": "recommended, shortest",
              "maxcellnodes": 5000
            }
          }
        },
        "maximum_intervals": 10,
        "maximum_locations": 2,
        "allow_compute_area": true
      },
      "routing": {
        "enabled": true,
        "mode": "normal",
        "routing_description": "This is a routing file from openrouteservice",
        "routing_name": "openrouteservice routing",
        "sources": [
          "data/osm_file.pbf"
        ],
        "init_threads": 1,
        "attribution": "openrouteservice.org, OpenStreetMap contributors",
        "elevation_preprocessed": false,
        "profiles": {
          "active": [
            "car"
          ],
          "default_params": {
            "encoder_flags_size": 8,
            "graphs_root_path": "data/graphs",
            "elevation_provider": "multi",
            "elevation_cache_path": "data/elevation_cache",
            "elevation_cache_clear": false,
            "instructions": true,
            "maximum_distance": 100000,
            "maximum_distance_dynamic_weights": 100000,
            "maximum_distance_avoid_areas": 100000,
            "maximum_waypoints": 50,
            "maximum_snapping_radius": 400,
            "maximum_avoid_polygon_area": 200000000,
            "maximum_avoid_polygon_extent": 20000,
            "maximum_distance_alternative_routes": 100000,
            "maximum_alternative_routes": 3,
            "maximum_distance_round_trip_routes": 100000,
            "maximum_speed_lower_bound": 80,
            "preparation": {
              "min_network_size": 200,
              "min_one_way_network_size": 200,
              "methods": {
                "lm": {
                  "enabled": true,
                  "threads": 1,
                  "weightings": "recommended,shortest",
                  "landmarks": 16
                }
              }
            },
            "execution": {
              "methods": {
                "lm": {
                  "disabling_allowed": true,
                  "active_landmarks": 8
                }
              }
            }
          },
          "profile-car": {
            "profiles": "driving-car",
            "parameters": {
              "encoder_flags_size": 8,
              "encoder_options": "turn_costs=true|block_fords=false|use_acceleration=false",
              "maximum_distance": 100000,
              "elevation": false,
              "maximum_snapping_radius": 350,
              "preparation": {
                "min_network_size": 200,
                "min_one_way_network_size": 200,
                "methods": {
                  "ch": {
                    "enabled": true,
                    "threads": 1,
                    "weightings": "fastest"
                  },
                  "lm": {
                    "enabled": false,
                    "threads": 1,
                    "weightings": "fastest,shortest",
                    "landmarks": 16
                  },
                  "core": {
                    "enabled": true,
                    "threads": 1,
                    "weightings": "fastest,shortest",
                    "landmarks": 64,
                    "lmsets": "highways;allow_all"
                  }
                }
              },
              "execution": {
                "methods": {
                  "ch": {
                    "disabling_allowed": true
                  },
                  "lm": {
                    "disabling_allowed": true,
                    "active_landmarks": 6
                  },
                  "core": {
                    "disabling_allowed": true,
                    "active_landmarks": 6
                  }
                }
              },
              "ext_storages": {
                "WayCategory": {},
                "HeavyVehicle": {},
                "WaySurfaceType": {},
                "RoadAccessRestrictions": {
                  "use_for_warnings": true
                }
              }
            }
          },
          "profile-hgv": {
            "profiles": "driving-hgv",
            "parameters": {
              "encoder_flags_size": 8,
              "encoder_options": "turn_costs=true|block_fords=false|use_acceleration=false",
              "maximum_distance": 100000,
              "elevation": false,
              "preparation": {
                "min_network_size": 200,
                "min_one_way_network_size": 200,
                "methods": {
                  "ch": {
                    "enabled": true,
                    "threads": 1,
                    "weightings": "recommended"
                  },
                  "lm": {
                    "enabled": true,
                    "threads": 1,
                    "weightings": "recommended,shortest",
                    "landmarks": 16
                  },
                  "core": {
                    "enabled": true,
                    "threads": 1,
                    "weightings": "recommended,shortest",
                    "landmarks": 64,
                    "lmsets": "highways;allow_all"
                  }
                }
              },
              "execution": {
                "methods": {
                  "ch": {
                    "disabling_allowed": true
                  },
                  "lm": {
                    "disabling_allowed": true,
                    "active_landmarks": 6
                  },
                  "core": {
                    "disabling_allowed": true,
                    "active_landmarks": 6
                  }
                }
              },
              "ext_storages": {
                "WayCategory": {},
                "HeavyVehicle": {
                  "restrictions": true
                },
                "WaySurfaceType": {}
              }
            }
          },
          "profile-bike-regular": {
            "profiles": "cycling-regular",
            "parameters": {
              "encoder_options": "consider_elevation=false|turn_costs=true|block_fords=false",
              "elevation": false,
              "ext_storages": {
                "WayCategory": {},
                "WaySurfaceType": {},
                "HillIndex": {},
                "TrailDifficulty": {}
              }
            }
          },
          "profile-bike-mountain": {
            "profiles": "cycling-mountain",
            "parameters": {
              "encoder_options": "consider_elevation=false|turn_costs=true|block_fords=false",
              "elevation": false,
              "ext_storages": {
                "WayCategory": {},
                "WaySurfaceType": {},
                "HillIndex": {},
                "TrailDifficulty": {}
              }
            }
          },
          "profile-bike-road": {
            "profiles": "cycling-road",
            "parameters": {
              "encoder_options": "consider_elevation=false|turn_costs=true|block_fords=false",
              "elevation": false,
              "ext_storages": {
                "WayCategory": {},
                "WaySurfaceType": {},
                "HillIndex": {},
                "TrailDifficulty": {}
              }
            }
          },
          "profile-bike-electric": {
            "profiles": "cycling-electric",
            "parameters": {
              "encoder_options": "consider_elevation=false|turn_costs=true|block_fords=false",
              "elevation": false,
              "ext_storages": {
                "WayCategory": {},
                "WaySurfaceType": {},
                "HillIndex": {},
                "TrailDifficulty": {}
              }
            }
          },
          "profile-walking": {
            "profiles": "foot-walking",
            "parameters": {
              "encoder_options": "block_fords=false",
              "elevation": false,
              "ext_storages": {
                "WayCategory": {},
                "WaySurfaceType": {},
                "HillIndex": {},
                "TrailDifficulty": {}
              }
            }
          },
          "profile-hiking": {
            "profiles": "foot-hiking",
            "parameters": {
              "encoder_options": "block_fords=false",
              "elevation": false,
              "ext_storages": {
                "WayCategory": {},
                "WaySurfaceType": {},
                "HillIndex": {},
                "TrailDifficulty": {}
              }
            }
          },
          "profile-wheelchair": {
            "profiles": "wheelchair",
            "parameters": {
              "encoder_options": "block_fords=true",
              "elevation": false,
              "maximum_snapping_radius": 50,
              "ext_storages": {
                "WayCategory": {},
                "WaySurfaceType": {},
                "Wheelchair": {
                  "KerbsOnCrossings": "true"
                },
                "OsmId": {}
              }
            }
          }
        }
      }
    },
    "logging": {
      "enabled": true,
      "level_file": "DEBUG_LOGGING.json",
      "location": "/var/log/ors",
      "stdout": true
    },
    "system_message": [
      {
        "active": false,
        "text": "This message would be sent with every routing bike fastest request",
        "condition": {
          "request_service": "routing",
          "request_profile": "cycling-regular,cycling-mountain,cycling-road,cycling-electric",
          "request_preference": "fastest"
        }
      },
      {
        "active": false,
        "text": "This message would be sent with every request for geojson response",
        "condition": {
          "api_format": "geojson"
        }
      },
      {
        "active": false,
        "text": "This message would be sent with every request on API v1 from January 2020 until June 2050",
        "condition": {
          "api_version": 1,
          "time_after": "2020-01-01T00:00:00Z",
          "time_before": "2050-06-01T00:00:00Z"
        }
      },
      {
        "active": false,
        "text": "This message would be sent with every request"
      }
    ]
  }
}

```

**Comment by sampaxk** [comment]
I'm experiencing the same issue, using the latest italian OSM file downloaded from Geofabrik.

I'm using the same `app.config` posted by @neatnerd .

This is the log: [log.txt](https://github.com/GIScience/openrouteservice/files/5684472/log.txt)

**Comment by nilsnolde** [comment]
No relation to Geofabrik here, it's simply the CGIAR download page being down since ages apparently. That's a recurring problem, we encountered it multiple times already.. And CGIAR keeps shifting their download URLs.. Probably time for
- either hosting the tiles independently (no idea about legality here)
- use alternative data sources: mapzen's tiles are open on amazon's open data buckets (though 1.6 TB for global dataset)..

Coming back to the problem:
@neatnerd 
>  in documentation it says that config is at /ors-conf

True, but if you look into the [docker-entrypoint script](https://github.com/GIScience/openrouteservice/blob/26562530d712869aaee82301eb12b866f69d206f/docker-entrypoint.sh#L24) you'll see that it'll be copied to the tomcat folder.

> Regarding global variable, I am not sure what you refer to

Right, there's indeed nothing in the default config.. I'm pretty sure though that there's supposed to be a `ors.services.routing.elevation` parameter you can set to `false`. AFAIK the profile-specific ones only determine whether or not elevation will be processed when generating the profile graph. If elevation is not set false globally it would still try to download them regardless what you set in the profiles.. Maybe @rabidllama or someone else can have a look.

Anyways, set `ors.services.routing.elevation_provider`  to `gmted` and it should work (will of course download and process that elevation source).

**Comment by Dzivo** [comment]
@nilsnolde 

> Anyways, set ors.services.routing.elevation_provider to gmted and it should work (will of course download and process that elevation source).

Can you explain how to set this parameter with an example i would appreciate it very much.

**Comment by nilsnolde** [comment]
You need to set the value of the JSON path `ors.services.routing.elevation_provider` in the `app.config` to `gmted`:

https://github.com/GIScience/openrouteservice/blob/9f278566c0af6978ca6ebca6125316006d936a08/openrouteservice/src/main/resources/app.config.sample#L81

BTW, if anyone would bother to address this open issue that wouldn't even be a problem and close at least #845 and this one:
https://github.com/GIScience/graphhopper/issues/34

**Comment by neatnerd** [comment]
> BTW, if anyone would bother to address this open issue that wouldn't even be a problem and close at least #845 and this one:
> [GIScience/graphhopper#34](https://github.com/GIScience/graphhopper/issues/34)

I have created PR #850, but I am not sure if this fixes it. I have tested it on my side and got mixed results, @nilsnolde can you verify, please?

**Comment by nilsnolde** [comment]
I'm not really part of this project anymore, but I can at least tell you that this, while nice of you to offer a PR, is not really what should be done. You can do that to your local copy of the app.config, but changing the defaults for everyone would be curing the symptoms not the cause:) and gmted is far inferior to CGIAR, so best is to have it back here:

The issue I pointed to in that GH fork is the one which needs fixing. You can look at this PR where I did the exact same a few years back (this keeps happening with CGIAR..): https://github.com/GIScience/graphhopper/pull/25

I guess the GH fork's branch to PR to is this one: https://github.com/GIScience/graphhopper/tree/ors_0.13

But might want to verify with the people who should actually answer issues like these;)

**Comment by neatnerd** [comment]
Alright, thanks for the response. Waiting until further comments in this thread. Please, note that until that point **the current version simply does not work** under any scenario as it does not start at all.

**Comment by nilsnolde** [comment]
Jep, absolutely, should have the highest priority.. I PRd: https://github.com/GIScience/graphhopper/pull/35. If you know how you can try to test. Failed for me for some reason, but really shouldn't..

**Comment by TheGreatRefrigerator** [comment]
@neatnerd , @nilsnolde sorry for the wait..

this should be fixed in 6.3.3 thanks to your pull request.
Feel free to reopen if it's not working.

**Comment by visaals** [comment]
FYI, if the latest patch 6.3.3 doesn't work, check to make sure you rebuild the image with `docker-compose build`. 

I was getting `Can't decode srtm_38_03.tif` errors even with the latest patched code until I rebuilt the images.. _sigh_ 2 hours I won't get back lol. Glad things work now! Thanks for the fix everyone 🙂
